File: codeforces/round478/C.cpp
#include <bits/stdc++.h>

typedef int64_t ll;
typedef uint64_t ull;

using namespace std;

#define mp make_pair
#define pb push_back
#define fi first
#define se second
#define lb lower_bound
#define ub upper_bound
#define el "\n"


int main(int argc, char const *argv[]) {
	ios_base::sync_with_stdio(false);
	cin.tie(NULL);
	#ifdef LOCAL_PROJECT
	freopen("in", "r", stdin);
	#endif

	ll n, q;
	cin >> n >> q;
	vector<ll> war(n);
	vector<ll> preCom(n);
	vector<ll> arr(n);
	for(auto &i : war)
		cin >> i;
	for(auto &i : arr)
		cin >> i;
	preCom[0] = war[0];
	for(ll i = 1; i < n; i++)
		preCom[i] = preCom[i-1] + war[i];
	ll totalHeal = preCom[n-1];
	ll curArr = 0;
	for(ll time = 0; time < q; time++){
		curArr += arr[time];
		if(curArr >= totalHeal){
			cout << n << el;
			curArr = 0;
		}
		else{
			auto idx = lb(preCom.begin(), preCom.end(), curArr);
			if(*idx - curArr == 0){
				idx++;
			}
			cout << n - (idx - preCom.begin()) << el;
		}
	}
	return 0;
}

File: codeforces/round478/C.py
def lb(arr, val):
	le = 0
	ri = len(arr)
	while le < ri:
		mid = (le+ri)//2
		if arr[mid] >= val:
			ri = mid
		else:
			le = mid+1
	return le

n, q = map(int, input().split(" "))
war = list(map(int, input().split(" ")))
arr = list(map(int, input().split(" ")))
pre = [war[0]]
for i in range(1, len(war)):
	pre.append(pre[i-1]+war[i])
totalHeal = pre[-1]
curArrw = 0
for i in range(q):
	curArrw += arr[i]
	if curArrw >= totalHeal:
		print(n)
		curArrw = 0
	else:
		idx = lb(pre, curArrw)
		if (pre[idx]-curArrw > 0):
			idx-=1
		idx += 1
		print(n - idx)
